### PR TITLE
Implement KV store

### PR DESCRIPTION
**Pull request by shrimetta**
**Problem Statement**
KV store

**Solution Overview**
Implemented KVStore with PillarDB 

Signed-off-by: Shri Bhargav Metta <shri.metta@seagate.com>

### CODE REVIEW

**Comment by shrimetta** [comment]
Closing this PR as it is stuck at Codacy/PR Quality review. Will reopen again.

File: py-utils/src/utils/store/dstore.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/python3
```
Comment (ujjwalpl) [inline py-utils/src/utils/store/dstore.py:1]: Please add a test py-utils/test/store/test_store.py

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/store/pillar.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    def get(self, key):
+        """Get pillar data for key."""
+
+        cmd = f"salt-call pillar.get {key} --out=json"
+        cmd_proc = SimpleProcess(cmd)
+        out, err, rc = cmd_proc.run()
+
+        if rc != 0:
+            msg = "get: salt-call: command not found" if rc == 127 else \
```
[the remark this answers is not in the document]
Comment (shrimetta) [reply py-utils/src/utils/store/pillar.py:39]: done

File: py-utils/src/utils/store/pillar.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                "get-pillar: Failed to get pillar data"
+            raise KvError(rc, msg)
+
+        res = json.loads(out)
```
[the remark this answers is not in the document]
Comment (shrimetta) [reply py-utils/src/utils/store/pillar.py:43]: done

File: py-utils/src/utils/store/pillar.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        res = res['local']
+
+        if not res:
```
[the remark this answers is not in the document]
Comment (shrimetta) [reply py-utils/src/utils/store/pillar.py:46]: In not found/error case res can be null or empty string and in that case we are throwing KvError

File: py-utils/src/utils/store/dstore.py (hunk 1 above)
Comment (shrimetta) [reply py-utils/src/utils/store/dstore.py:1]: done

**Pull request closed without merging**